### ISSUE TITLE
improve usability of SSL handshake exception

### ISSUE THREAD

**Issue by wfurt** [dotnet/corefx#34905]
There is growing number of cases (like #32976) where uses see something like:

```
---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure.
   at System.Net.Security.SslState.StartSendAuthResetSignal(ProtocolToken message, AsyncProtocolRequest asyncRequest, ExceptionDispatchInfo exception)
```

From that, it is very difficult to diagnose real root cause 
It would be nice if the exception can provide more specific reason. This may not be complete list but it would be nice to distinguish following cases:
- wrong time
- unknown/untrusted CA
- name mismatch
- CRL/OSCP revocation issues
- custom verification callback denial
- extended key usage violation or other attribute problems

**Comment by karelz** [comment]
@wfurt just one area label please ;)

**Comment by weitzhandler** [comment]
Any workarounds?
How can we avoid that exception on a development machine?
Since I'm working on a tenant app, I'm using fake full URLs configured in the hosts file.

I'm using IdentityServer4 to generate JWT tokens, when trying to access resources, I get the exception.

**Comment by wfurt** [comment]
I don't understand your question @weitzhandler. This issue is about improving text but it will not change the behavior anyhow. 
If you are failing to connect, you need to check your trust chain. Alternatively, you can can register validation delegate and process it your self.

**Comment by weitzhandler** [comment]
Yes there were issues with the certificates. I got it to work now.

**Comment by weitzhandler** [comment]
I'm deleting my previous comments (including this one) to keep this issue clean.

**Comment by karelz** [comment]
One concrete example is in #37599
